packages/flutter_tools/lib/src/isolated/sdk_web_configuration.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:async';
+
+import 'package:dwds/dwds.dart';
+
+import '../artifacts.dart';
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+
+/// Provides paths to SDK files for dart SDK used in flutter.
+class SdkWebConfigurationProvider extends SdkConfigurationProvider {
+  SdkWebConfigurationProvider();
+  SdkConfiguration _configuration;
+
+  /// Create and validate configuration matching the default SDK layout.
+  /// Create configuration matching the default SDK layout.
+  @override
+  Future<SdkConfiguration> get configuration async {
+    if (_configuration == null) {
+      final String sdkDir = globals.artifacts.getHostArtifact(HostArtifact.flutterWebSdk).path;
```

Comment:
Could you pass in `artifacts` via the constructor, rather than grab it from globals? We are trying to reduce the overall use of globals in the tool codebase.

Reply:
I am not very familiar with flutter tools code, could you please point me to some code where I can get the artifacts in a approved way?

Reply:
Here's an example: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/analyze_base.dart#L50

We'll still be using the same `globals.artifacts` instance that we provide when constructing this object in the tool, but by direct injecting it in the constructor it reduces the chance of a global instance leaking into our unit tests.

Reply:
For example, I don't think you would need the `context.run()` in `packages/flutter_tools/test/web.shard/sdk_web_configuration_test.dart` if the test artifacts are passed in via the constructor

Reply:
got it, thanks!

Reply:
done!